plugins/builds/index.js

```diff
@@ -720,17 +669,39 @@ async function createOrRunNextBuild({ buildFactory, jobFactory, eventFactory, pi
     };
 
     /* CHECK WHETHER NEXT BUILD EXISTS */
-    const nextBuild = await getNextBuild({
-        isExternal,
-        pipelineFactory,
-        externalPipelineId,
-        externalJobName,
-        jobFactory,
-        eventFactory,
-        event,
-        workflowGraph,
-        nextJobName
-    });
+    let nextBuild;
+
+    // If next build is external, return the latest build with same job ID
+    if (isExternal) {
+        const p = await pipelineFactory.get(externalPipelineId);
+        const jobArray = await p.getJobs({ params: { name: externalJobName } });
+        const j = await jobFactory.get(jobArray[0].id);
+
+        nextBuild = await j.getLatestBuild({ status: 'CREATED' });
+    }
+    // Get finished internal builds from event
+    const finishedInternalBuilds = await getFinishedBuilds(event, eventFactory);
+
+    if (event.parentEventId) {
+        Object.keys(parentBuilds).forEach((pid) => {
+            parentBuilds[pid].eventId = event.id;
+            Object.keys(parentBuilds[pid].jobs).forEach((jName) => {
+                if (parentBuilds[pid].jobs[jName] === null) {
+                    const jobId = workflowGraph.nodes.find(node =>
+                        node.name === trimJobName(jName)).id;
+
+                    parentBuilds[pid].jobs[jName] = finishedInternalBuilds.find(b =>
+                        b.jobId === jobId).id;
+                }
+            });
+        });
+    }
+
+    // If next build is internal, look at the finished builds for this event
+    const jobId = workflowGraph.nodes.find(node =>
+        node.name === trimJobName(nextJobName)).id;
+
+    nextBuild = finishedInternalBuilds.find(b => b.jobId === jobId);
```

Comment:
I don't see an `else` wrap here, so this `nextBuild` will over write `external` logic from line `675`

Reply:
fixed